jina/executors/indexers/__init__.py

```diff
@@ -196,32 +196,13 @@ def flush(self):
         except:
             pass
 
-    def _filter_nonexistent_keys_values(self, keys: Iterable, values: Iterable, existent_keys: Iterable,
-                                        check_path: str) -> Tuple[List, List]:
-        keys = list(keys)
-        values = list(values)
-        if len(keys) != len(values):
-            raise ValueError(f'Keys of length {len(keys)} did not match values of length {len(values)}')
-        indices_to_drop = self._get_indices_to_drop(keys, existent_keys, check_path)
-        keys = [keys[i] for i in range(len(keys)) if i not in indices_to_drop]
-        values = [values[i] for i in range(len(values)) if i not in indices_to_drop]
-        return keys, values
-
-    def _filter_nonexistent_keys(self, keys: Iterable, existent_keys: Iterable, check_path: str) -> List:
-        keys = list(keys)
-        indices_to_drop = self._get_indices_to_drop(keys, existent_keys, check_path)
-        keys = [keys[i] for i in range(len(keys)) if i not in indices_to_drop]
-        return keys
-
-    def _get_indices_to_drop(self, keys: List, existent_keys: Iterable, check_path: str):
-        indices_to_drop = []
-        for key_index, key in enumerate(keys):
-            if key not in existent_keys:
-                indices_to_drop.append(key_index)
-        if indices_to_drop:
-            self.logger.warning(
-                f'Key(s) {[keys[i] for i in indices_to_drop]} were not found in {check_path}. Continuing anyway...')
-        return indices_to_drop
+    def _filter_nonexistent_keys_values(self, keys: Iterable, values: Iterable, existent_keys: Iterable) -> Tuple[
+        Iterable, Iterable]:
+        filtered_list = [[key, value] for key, value in zip(keys, values) if key in existent_keys]
```

Comment:
Can it be better to return a Tuple? for inmutability?

Reply:
do you mean doing:
```
tuple([key_value[0] for key_value in filtered_list])
```
I can do it. An argument against this would be that we use lists almost everywhere and no tuples.
What do you think?

Reply:
no tuple, just using ( ) would make it inmutable right? not important so forget about it

Reply:
using () we would create a generator object. Actually, I started with just using generators here to make it super performant. But then I realized that the vector indexer converts everything into numpy array anyways.
So we have to call `list` on the key generator which is not so nice: `np.array(list(keys), (np.str_, self.key_length))`

Reply:
I believe he meant `[(key, value)...]`, not?